community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -0,0 +1,385 @@
+    private final Path configDir;
+
+    static long recommendOsMemory( long totalMemoryBytes )
+    {
+        Brackets brackets = findMemoryBrackets( totalMemoryBytes );
+        return brackets.recommend( Bracket::osMemory );
+    }
+
+    static long recommendHeapMemory( long totalMemoryBytes )
+    {
+        Brackets brackets = findMemoryBrackets( totalMemoryBytes );
+        return brackets.recommend( Bracket::heapMemory );
+    }
+
+    static long recommendPageCacheMemory( long totalMemoryBytes )
+    {
+        long osMemory = recommendOsMemory( totalMemoryBytes );
+        long heapMemory = recommendHeapMemory( totalMemoryBytes );
+        long recommendation = totalMemoryBytes - osMemory - heapMemory;
+        recommendation = Math.max( mebiBytes( 100 ), recommendation );
+        recommendation = Math.min( tebiBytes( 16 ), recommendation );
+        return recommendation;
+    }
+
+    private static Brackets findMemoryBrackets( long totalMemoryBytes )
+    {
+        double totalMemoryGB = ((double) totalMemoryBytes) / ((double) gibiBytes( 1 ));
+        Bracket lower = null;
+        Bracket upper = null;
+        for ( int i = 1; i < datapoints.length; i++ )
+        {
+            if ( totalMemoryGB < datapoints[i].totalMemory )
+            {
+                lower = datapoints[i - 1];
+                upper = datapoints[i];
+                break;
+            }
+        }
+        if ( lower == null )
+        {
+            lower = datapoints[datapoints.length - 1];
+            upper = datapoints[datapoints.length - 1];
+        }
+        return new Brackets( totalMemoryGB, lower, upper );
+    }
+
+    public static Arguments buildArgs()
+    {
+        return new Arguments()
+                .withArgument( new OptionalNamedArg( ARG_MEMORY, "8g", OsBeanUtil.getTotalPhysicalMemory() + "",
+                        "Recommend memory settings with respect to the given amount of memory, " +
+                        "instead of the total memory of the system running the command." ) )
+                .withArgument( new OptionalBooleanArg( ARG_PRINT_CONF, false,
+                        "Print what an updated neo4j.conf file would look like to standard out. Does not change " +
+                        "existing settings unless --overwrite is specified." ) )
+                .withArgument( new OptionalBooleanArg( ARG_UPDATE_CONF, false,
+                        "Update the neo4j.conf file, or the given --conf-file, in-place by adding the recommended " +
+                        "memory settings, unless they are already specified. " +
+                        "Specify --overwrite to also update existing settings." ) )
+                .withArgument( new OptionalBooleanArg( ARG_OVERWRITE, false,
+                        "Causes --print-conf and --update-conf to overwrite existing settings, if set." ) )
+                .withArgument( new OptionalNamedArg( ARG_CONF_FILE, "conf" + File.separator + "neo4j.conf", "",
```

Comment:
One reason for me saying so is that on Linux systems the user running the command will typically not have the same environment set as the Neo4j service does. So there will be no way to figure out where the config is.

A user will be invoking it like this

    neo4j-admin memrec .... /etc/neo4j/neo4j.conf

that command will fail because the user doesn't have write access to `/etc`. So second time around the user sudos:

    sudo neo4j-admin memrec .... /etc/neo4j/neo4j.conf

In neither case does the user have `NEO4J_CONF` set in their environment because that is set in the neo4j system service. So there is no way for the tool to figure out where to put the config in any reasonable automatic sense.